laurafitzgerald/app-services-cli
# [WIP] Poc yaml gen gitops

WIP
Note: I commited apis as there were some dependency issues. These can be resolved later but I didn't want it to block code samples. If we go with this approach it is my opinion that apis should be moved to a central location away from their code base so as to avoid the dependency issues faced by operators genrally. 

WHAT
Automate the steps mentioned here -> https://access.redhat.com/documentation/en-us/red_hat_3scale_api_management/2.10/html/getting_started/first-steps-with-threescale_configuring-your-api  for a user to run a simple cli command to generate yaml that would result in the same configuration in 3scale as described

## Verification Steps
Run ` go run cmd/rhoas/main.go rhoam generate --name=test --url=www.example.com --env=staging`

Verify that there is a test directory created with two files
- backend.yaml
- product.yaml

with the appropriate configuration based on the parameters passed. 

TODO
Add a sub command to generate e.g. apis tenants etc. to allow for all use cases we want to support via the cli
Decide on whether this approach removes the need to have a top level yaml which gitops could consume and present to the cluster in the appropriate 3scale readable crs. 




### Next Steps
https://github.com/3scale/3scale-operator/pull/778
